lib/chef/provider/package/zypper.rb

```diff
@@ -45,11 +46,12 @@ def get_versions(package_name)
               is_installed = true
               logger.trace("#{new_resource} is installed")
             when /^Status *: out-of-date \(version (.+) installed\) *$/
+              out_of_date = true
               current_version = $1.strip
               logger.trace("#{new_resource} out of date version #{current_version}")
             end
           end
-          current_version = candidate_version if is_installed
+          current_version = candidate_version if is_installed && !out_of_date
```

Comment:
didn't you just make this fail on the `:install` action -- because now if you have a version installed which is out of date, the current_version will be nil, which will cause the package to get installed (aka upgraded)?

Reply:
oh I think I see what the API is here.  this should probably just be:

```
current_version ||= candidate_version if is_installed
```

then you can eliminate the `out_of_date` flag and instead rely on the current_version being nil or not.